config/routes.rb

```diff
@@ -246,7 +246,11 @@
 
     resources :permissions, :only => [:index]
 
-    resources :auth_source_ldaps, :except => [:show]
+    resources :auth_source_ldaps, :except => [:show] do
+      collection do
+        put 'test_connection'
```

Comment:
I'd say use the API endpoint on the form and remove the route and method from the regular controller if you can. Is there any specific reason you want it in the regular controller? Also use GET if you're not modifying any resource - like in this case (I know Foreman doesn't always do this, unfortunately :cry: )